pkg/helpers/tokencmd/basicauth.go

```diff
@@ -14,6 +14,18 @@ import (
 	"github.com/openshift/oc/pkg/helpers/term"
 )
 
+// BasicNoUsername enables BasicNoUsernameError implementation
+type BasicNoUsername struct{}
+
+func (e *BasicNoUsername) Error() string {
+	return "did not receive username - print token url instead of basic prompt"
+}
+
+// BasicNoUsernameError returns an error for a basic challenge without a username
+func BasicNoUsernameError() error {
```

Comment:
`NewBasicAuthNoUsernameError()`

Reply:
yes, much better, updated : )

Also, in request_token.go, updated `BasicNoUsernameMessage` to `BasicAuthNoUsernameMessage`